ibis/expr/tests/test_geospatial.py

```diff
@@ -1,66 +1,147 @@
 
-    polygon1 = (
-        (0, 0), (1, 0), (0.5, 1), (0, 0)
-    )
-    polygon2 = (
-        ((35, 10), (45, 45), (15, 40), (10, 20), (35, 10)),
-        ((20, 30), (35, 35), (30, 20), (20, 30))
-    )
-    multipolygon = [polygon1, polygon2]
-    ibis.literal(multipolygon, type='multipolygon{}'.format(modifier))
+pytestmark = pytest.mark.skipif(
+    sys.version_info < (3, 6),
+    reason='Geo Spatial support available just for Python >= 3.6',
+)
 
+pytest.importorskip('geoalchemy2')
+pytest.importorskip('shapely')
+pytest.importorskip('geopandas')
 
-def test_geo_ops_smoke(geo_table):
-    """Smoke tests for geo spatial operations."""
-    t = geo_table
+pytest.mark.postgis
+pytest.mark.omniscidb
 
-    # alias for fields
-    point = t.geo_point
-    linestring = t.geo_linestring
-    polygon = t.geo_polygon
-    multipolygon = t.geo_multipolygon
+mock_omniscidb = GeoMockConnectionOmniSciDB()
+mock_postgis = GeoMockConnectionPostGIS()
 
-    # test ops
-    point.srid()
-    point.set_srid(4326)
-    point.x()
-    point.y()
 
-    linestring.contains(point)
-    linestring.end_point()
-    linestring.length()
-    linestring.max_distance(point)
-    linestring.point_n(1)
-    linestring.start_point()
-    linestring.x_max()
-    linestring.x_min()
-    linestring.y_max()
-    linestring.y_min()
+@pytest.mark.parametrize('backend', [mock_postgis])
+@pytest.mark.parametrize(
+    'modifier', ['', ';4326', ';4326:geometry', ';4326:geography']
+)
+@pytest.mark.parametrize(
+    'shape,value,expected',
+    [
+        ('point', (30, 10), 'POINT (30 10)'),
+        (
+            'linestring',
+            ((30, 10), (10, 30), (40, 40)),
+            'LINESTRING (30 10, 10 30, 40 40)',
+        ),
+        (
+            'polygon',
+            (
+                ((35, 10), (45, 45), (15, 40), (10, 20), (35, 10)),
+                ((20, 30), (35, 35), (30, 20), (20, 30)),
+            ),
+            (
+                'POLYGON ((35 10, 45 45, 15 40, 10 20, 35 10), '
+                + '(20 30, 35 35, 30 20, 20 30))'
+            ),
+        ),
+        (
+            'polygon',
+            ((30, 10), (40, 40), (20, 40), (10, 20), (30, 10)),
+            'POLYGON ((30 10, 40 40, 20 40, 10 20, 30 10))',
+        ),
+        (
+            'multipolygon',
+            (
+                ((40, 40), (20, 45), (45, 30), (40, 40)),
+                (
+                    (
+                        (20, 35),
+                        (10, 30),
+                        (10, 10),
+                        (30, 5),
+                        (45, 20),
+                        (20, 35),
+                    ),
+                    ((30, 20), (20, 15), (20, 25), (30, 20)),
+                ),
+            ),
+            (
+                'MULTIPOLYGON (((40 40, 20 45, 45 30, 40 40)), '
+                + '((20 35, 10 30, 10 10, 30 5, 45 20, 20 35), '
+                + '(30 20, 20 15, 20 25, 30 20)))'
```

Comment:
Same comment as above, though if you like this style more I'm fine with that.

Reply:
yeah .. I have used both .. but I like more the explicit